### ISSUE TITLE
Android and iOS support? 

### ISSUE THREAD

**Issue by AlGrenadine** [jcelerier/RtMidi17#4]
Title says it all :)

**Comment by jcelerier** [comment]
hello ! that would be nice - I would expect that it "just works" on android as Android uses ALSA behind the scene if I'm not mistaken but I do not knjow at all on iOS - is it not CoreMidi there too ?

**Comment by AlGrenadine** [comment]
Yes coremidi seems to work on iOS but alsa doesn't on Android afaik

**Comment by jcelerier** [comment]
damn I hope we don't have to call Java APIs to make it work

**Comment by AlGrenadine** [comment]
I fear you have to :(

**Comment by jcelerier** [comment]
just checked and apparently no ! android NDK comes with a C/C++ midi API:

https://developer.android.com/ndk/reference/group/midi

**Comment by atsushieno** [comment]
In case it matters, Android Native MIDI API is available only in Android 10 or later.

**Comment by jcelerier** [comment]
argh :) communicating through JNI is an... exercice in patience to say the least (and I have very few experience with it).
I'd be ok with trying to do it with the more recent API, then if someone wants to contribute support for the java-based API it'll be 100 % welcome but I don't have the courage for it

**Comment by jcelerier** [comment]
50% done, I could make the library build for iOS, there were a couple minor things.

**Comment by jcelerier** [comment]
the code that seems required for android is .... yuck yuck yuck

https://github.com/thestk/rtmidi/blob/master/RtMidi.cpp#L4838

**Comment by AlGrenadine** [comment]
> the code that seems required for android is .... yuck yuck yuck
> 
> https://github.com/thestk/rtmidi/blob/master/RtMidi.cpp#L4838

Excellent :)

**Comment by jcelerier** [comment]
https://github.com/android/ndk-samples/blob/main/native-midi/app/src/main/cpp/AppMidiManager.cpp 

definitely not ahah, I'd rather like to avoid having some weird JNI calls in the middle. Maybe it can be part of some example though - the end-user would be able to implement these calls however they see fit.

**Comment by AlGrenadine** [comment]
Anyway, now it's implemented in rtmidi it's still good news :)
